ui/app/components/app-form/project-settings.hbs

```diff
@@ -0,0 +1,163 @@
+            value={{this.project.dataSource.git.ssh.user}}
+            {{on "input" (fn this.setSshAuth "user")}}
+          />
+        </div>
+        <div class="pds-formField">
+          <label for="git-source-ssh-password" class="pds-fieldName">
+            Passphrase (optional)
+          </label>
+          <Pds::Input
+            @type="password"
+            id="git-source-ssh-password"
+            placeholder="**********"
+            value={{this.project.dataSource.git.ssh.password}}
+            {{on "input" (fn this.setSshAuth "password")}}
+          />
+        </div>
+        <div class="pds-formField">
+          <label for="git-source-ssh-key" class="pds-fieldName">
+            SSH Key
+          </label>
+          <Pds::Textarea
+            id="git-source-ssh-key"
+            placeholder="*****************"
+            value={{this.git.ssh.privateKeyPem}}
+            required={{this.authSSH}}
+            {{on "input" (fn this.setSshAuth "privateKeyPem")}}
+          />
+        </div>
+      {{else if this.authBasic}}
+        <div class="pds-formField">
+          <label for="git-source-username" class="pds-fieldName">
+            Username
+          </label>
+          <Pds::Input
+            @type="text"
+            id="git-source-username"
+            placeholder="gitusername"
+            required={{this.authBasic}}
+            value={{this.project.dataSource.git.basic.username}}
+            {{on "input" (fn this.setBasicAuth "username")}}
+          />
+        </div>
+        <div class="pds-formField">
+          <label for="git-source-password" class="pds-fieldName">
+            Password
+          </label>
+          <Pds::Input
+            @type="password"
+            id="git-source-password"
+            placeholder="**********"
+            required={{this.authBasic}}
+            value={{this.project.dataSource.git.basic.password}}
+            {{on "input" (fn this.setBasicAuth "password")}}
+          />
+        </div>
+      {{/if}}
+
+      <div class="pds-formField">
+        <label class="pds-fieldName">
+          Automated Sync
+        </label>
+        <div class="pds-formFieldToggle">
+          <XToggle
+            @showLabels={{true}}
+            @name="git-sync-enabled"
+            @value={{this.project.dataSourcePoll.enabled}}
+            @size="small"
+            @onToggle={{fn (mut this.project.dataSourcePoll.enabled)}} as |toggle|>
+            <toggle.switch
+              @onLabel='git sync on'
+              @offLabel='git sync off'
+              @name="Git Sync"
+              />
+
+            <toggle.label for="git-sync-enabled">
+              <span class="pds-toggleLabel">
+                Enable automated deployment on changes
+              </span>
+            </toggle.label>
+          </XToggle>
+
+        </div>
+      </div>
+    </fieldset>
+  </div>
+
+  <Pds::Button @variant="primary" {{on "click" this.saveSettings}}>Apply</Pds::Button>
```

Comment:
If you move the save handler to the form tag at the very top with:

`<form class="pds-form" {{on "submit" this.saveSettings}}>` 

and then change this to be a `@type="submit"` button, you get the added nicety of hitting "enter" to submit the form if any of the fields are focused.

Also something to think about here: the "saving" state isn't being handled so it's possible for a user who's click-happy to trigger save more than once.

In HCP we use `ember-concurrency` tasks for all saves so that we can use the derived state to disable the save buttons too (certainly not required, but if you're using e-c otherwise, it's easier than tracking the save state manually).

Reply:
done: https://github.com/hashicorp/waypoint/pull/1057/commits/f20301d89246e6ee9d57e0679570cbd513266143